src/null-safety/faq.md

```diff
@@ -301,6 +301,77 @@ and situations where a null value is really expected. So the tool tells you what
 it knows ("it looks like this condition will always be false!") and lets you
 decide what to do.
 
+## What should I know about dart2js and null safety?
+
+For a long time, the dart2js compiler has had optimizations specifically
+targeting null values and null checks. Because of that, null safety is not
+expected to affect much the output of the compiler.
+
+A few notes that are worth highlighting:
+
+* dart2js emits `!` null assertions, but you may not notice them.
+  That's because pre null-safety dart2js already emitted null checks
+  (so they are already in the existing programs).
+
+* dart2js emits these null assertions regardless of sound/unsound null safety
+  and regardless of optimization level. In fact, dart2js doesn't remove
+  `!` when using `-O3` or `--omit-implicit-checks`.
+
+* dart2js may optimize away unnecessary null checks. This is because the same
+  optimizations done by dart2js in the past will be able to eliminate
+  unnecessary null checks when it knows the value is not null.
+
+* By default dart2js would generate parameter subtype checks (runtime checks
+  used to ensure covariant virtual calls are given appropriate arguments).
+  These are elided with the `--omit-implicit-checks` flag just as before.
```

Comment:
maybe we can say:

 > ... with the `-O3` and `-O4` flags just as before.

In our internal build systems we use the `--omit-implicit-checks` flags, which is why I used that for the internal FAQ, but external users are more likely to be familiar with the `-O` flags.

Reply:
Good point. On the website we don't document the flag `--omit-implicit-checks` but rather mention it as part of the `-O` flags.